src/game/client.cpp

```diff
@@ -1147,8 +1147,16 @@ namespace client
             defformatstring(st, " (to team %s)", game::colourteam(f->team));
             concatstring(name, st);
         }
-        if(flags&SAY_ACTION) formatstring(line, "\fv* %s %s", name, msg);
-        else formatstring(line, "\fw<%s> %s", name, msg);
+        else if (flags&SAY_WARNING)
+        {
+            defformatstring(swarn, "%s\fzwy, ", game::colourname(t));
+            prependstring(msg, swarn);
```

Comment:
This would have been way easier with `std::string`, we need to find time for that someday.